### PR TITLE
[Bugfix][Spec Decode] Seed both-XD-RoPE drafting from the xdrope positions buffer

### LINKED ISSUE #54555: [Bug]: V1 spec-decode proposer never constructs the positions buffer a both-XD-RoPE drafter seeds from
### 🐛 Describe the bug

The V1 spec-decode proposer crashes on the first sequential draft step for any both-XD-RoPE configuration (target model `uses_xdrope_dim > 0` and draft model `uses_xdrope_dim > 0`), because the position-buffer init and the draft-loop seed disagree about which buffer this config uses.

**Buffer init** (`vllm/v1/spec_decode/llm_base_proposer.py`, `SpecDecodeBaseProposer.__init__`):

```python
if self.uses_mrope:
    self.mrope_positions = torch.zeros(...)
elif self.uses_xdrope_dim > 0 and self.draft_uses_xdrope_dim > 0:
    self.xdrope_positions = torch.zeros(...)
else:
    self.positions = torch.zeros(...)
```

For both-XD-RoPE, `self.positions` is never constructed.

**Draft-loop seed** (`propose()`, before the `num_speculative_tokens - 1` loop):

```python
if self.uses_mrope:
    positions = self.mrope_positions[:, token_indices_to_sample]
else:
    positions = self.positions[token_indices_to_sample]
```

Only `uses_mrope` gets a dedicated branch, so a both-XD-RoPE run lands in the `else` and reads `self.positions`, which does not exist for this config: immediate `AttributeError` on the first speculative step. The same mismatch exists in `_get_positions` / `_set_positions`, which do handle the both-XD-RoPE branch, so the buffers that ARE written are never the ones the seed reads.

**Reachability.** No model in main implements `get_xdrope_input_positions` yet, so nothing in-tree can trigger this today; it is a latent crash on the landing path. The XD-RoPE machinery is clearly mid-landing (scheduler, runner, `RopeState`, #54232), the runner comment names HunYuan-VL as the intended family, and the first both-XD-RoPE drafter with `num_speculative_tokens >= 2` (sequential, non-parallel) will hit this on its very first proposal. Worth fixing before that model PR lands rather than inside it.

**Fix shape.** Treat both-XD-RoPE the way M-RoPE is treated in the seed and in `_update_positions_dependent_metadata`: seed from `self.xdrope_positions[:, token_indices_to_sample]` and take dim 0 as the slot coordinate. That is sound because XD-RoPE decode positions are the absolute token index on every dim by construction: `get_rope_state` builds the XD-RoPE state with `has_delta=False`, so `_prepare_rope_positions_kernel` uses `orig_pos` for all dims during decode. A CPU-only proposer-level test with synthetic XD-RoPE positions can pin this without a checkpoint; I am happy to open that PR if the direction sounds right.

**Relation to #54498.** Found while analyzing that issue. Different root: #54498 is a wrong-but-present coordinate for M-RoPE, this one is a missing buffer for XD-RoPE. The two fixes touch adjacent branches of the same function and should not conflict.

**Disclosure.** This analysis was done with AI assistance; every claim above was verified against current main (`810bc32`) by reading the cited code paths.

### PR DESCRIPTION
**Pull request by he-yufeng**
Fixes #54555.

## Purpose

The both-XD-RoPE config (target model `uses_xdrope_dim > 0` and draft model `uses_xdrope_dim > 0`) allocates `xdrope_positions` instead of the 1D `positions` buffer, but the draft loop seeds `positions` from `self.positions` for anything that is not `uses_mrope`, so the first both-XD-RoPE drafter with `num_speculative_tokens >= 2` raises `AttributeError` on its first speculative step. The seed now reads from `xdrope_positions` for that config, and `_update_positions_dependent_metadata` takes the slot coordinate from dim 0 for both multi-dim flavors: XD-RoPE decode positions are the absolute token index on every dim by construction (`get_rope_state` builds XD-RoPE with `has_delta=False`, so `_prepare_rope_positions_kernel` uses `orig_pos`), which is exactly what the slot math needs.

No model in main implements `get_xdrope_input_positions` yet, so this path cannot be exercised end-to-end today; the fix lands ahead of the first XD-RoPE drafter (HunYuan-VL family per the runner comment) instead of inside it. Non-XD-RoPE paths are byte-for-byte unchanged.

## Test Plan

New `tests/v1/spec_decode/test_xdrope_seed_positions.py`: the seed reads the xdrope buffer for the both-XD-RoPE config (and never touches the missing 1D buffer), the M-RoPE and plain paths keep their buffers, `_uses_multidim_positions` covers both flavors and excludes a target-xdrope/non-xdrope-draft mix, and `_update_positions_dependent_metadata` feeds dim 0 to the slot kernel for the both-XD-RoPE config (kernel stubbed, CPU-only).

## Verification

- `.venv/bin/python -m pytest tests/v1/spec_decode/test_xdrope_seed_positions.py tests/v1/spec_decode/test_llm_base_proposer.py -q` -> 8 passed on current main (`810bc32`).
- `ruff check` and `ruff format --check` clean on both files.
- Not run: end-to-end drafting with an XD-RoPE checkpoint (none exists in main yet); the fix is exercised at the proposer state level above.

This change was prepared with AI assistance (Kimi K3); the crash mechanism was verified by reading the buffer init, the seed path, and the XD-RoPE decode position contract before writing the fix.

### CODE REVIEW

**Comment by he-yufeng** [comment]
Rebased onto current main. The branch previously carried the Eagle3 aux-count commit from #54598, so the diff showed two unrelated fixes; it is now just the XD-RoPE seeding change. Also dropped a duplicated `hidden_states[token_indices_to_sample]` gather that sneaked in during the original stack.

`tests/v1/spec_decode/test_xdrope_seed_positions.py`: 5 passed locally, ruff check and format clean.

**Comment by he-yufeng** [comment]
Still applies on current main: `llm_base_proposer.py:637-640` seeds `positions = self.positions[...]` for every non-M-RoPE drafter, but the both-XD-RoPE branch at :192-197 allocates `xdrope_positions` instead of `positions`, so the first draft step raises AttributeError; `:787` likewise never extracts dim 0 for the XD-RoPE flavor. The five-test CPU suite in this PR still passes on the current base. One honest caveat: no model on main implements `get_xdrope_input_positions` yet, so this is a forward fix ahead of the HunYuan-VL drafters rather than a live crasher today.
